### ISSUE TITLE
WiFi not working

### ISSUE THREAD

**Issue by vinodhmoodley** [marcosfad/mbp-ubuntu#4]
I have a 15,2 MacBook Pro:

`RequestedFiles" = (\{"Firmware"="C-4364__s-B2/maui.trx","TxCap"="C-4364__s-B2/maui-X3.txcb","Regulatory"="C-4364__s-B2/maui-X3.clmb","NVRAM"="C-4364__s-B2/P-maui-X3_M-HRPN_V-m__m-7.7.txt"\})\
    | |   |       | |       "images" = \{"C-4364__s-B2/maui-X3.txcb"=\{"imagetype"="TxCap","required"=No,"imagename"="C-4364__s-B2/maui-X3.txcb"\},"C-4364__s-B2/maui-X3.clmb"=\{"imagetype"="Regulatory","required"=Yes,"imagename"="C-4364__s-B2/maui-X3.clmb"\},"C-4364__s-B2/maui.trx"=\{"imagetype"="Firmware","required"=Yes,"imagename"="C-4364__s-B2/maui.trx"\},"C-4364__s-B2/P-maui-X3_M-HRPN_V-m__m-7.7.txt"=\{"imagetype"="NVRAM","required"=Yes,"imagename"="C-4364__s-B2/P-maui-X3_M-HRPN_V-m__m-7.7.txt"\}\}\`

 and I get the following error if I use the downloaded Mojave firmware:

`brcmf_pcie_download_fw_nvram: FW failed to initialize`

If I try to run:

`modprobe -r brcmfmac; modprobe brcmfmac rambase_addr=0x160000`

I get the following error:
`modprobe: ERROR: ../libkmod/libkmod-module.c:799 kmod_module_remove_module() could not remove 'brcmfmac': Operation not permitted`

I'm not sure why but everyone else who uses this method of getting wifi working don't get this "operation not permitted" error. I'm not sure what else to do besides starting from scratch and using a clean Ubuntu 20.04 ISO. Everything else eg. keyboard, trackpad, touchbar etc works besides WiFi.

**Comment by gamunu** [comment]
@vinodhmoodley try packages from https://packages.aunali1.com/apple/wifi-fw/ . The firmware that came with Catalina did not work for me either.

**Comment by vinodhmoodley** [comment]
> @vinodhmoodley try packages from https://packages.aunali1.com/apple/wifi-fw/ . The firmware that came with Catalina did not work for me either.

Thanks for trying to help but as mentioned in my post, I did use that Mojave firmware and not the Catalina version. In the mean time, I've installed Fedora and it did the same thing. I've re-installed Ubuntu several more times and it does the same thing. I asked for help on the Linux on MBP 2018+ Discord and one other user has the same issue as me.

**Comment by vinodhmoodley** [comment]
I got the WiFi issue fixed. I used a newer build of your mbp-ubuntu iso and installed like normal. I then copied the firmware files exactly like what I did previously and now it works. I'm not sure what changed but somehow or other, it's working despite following the exact same method I used before.

However, if i try to connect to a WiFi network, any WiFi network, it keeps asking for a password even though the password I enter is definitely correct. Turns out that Network Manager needs IWD instead of WPA Supplicant to resolve this isssue. I installed IWD and edited the Network Manager config to use IWD as it's back end and it now works. 

I detailed what I did here if anyone else has this issue:
https://vinodhsblog.co.za/ubuntu-20-04-lts-on-a-macbook-pro/
